gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts

```diff
@@ -26,32 +26,30 @@ export const wrapCommand = (command) => {
       logger.info(`Running ${command.id} command using Serum Multisig`)
 
       this.command = new command({ ...flags, bufferSize: MAX_BUFFER_SIZE }, args)
-      this.command.invokeMiddlewares(this.command, this.command.middlewares)
       this.require(!!process.env.MULTISIG_ADDRESS, 'Please set MULTISIG_ADDRESS env var')
       this.multisigAddress = new PublicKey(process.env.MULTISIG_ADDRESS)
     }
 
     execute = async () => {
+      // TODO: Command underneath will try to load its own provider and wallet if invoke middlewares, but we should be able to specify which ones to use, in an obvious better way
+      this.command.provider = this.provider
+      this.command.wallet = this.wallet
+
       const multisig = getContract(CONTRACT_LIST.MULTISIG, '')
       this.program = this.loadProgram(multisig.idl, multisig.programId.toString())
 
-      // Falling back on default wallet if signer is not provided or execute flag is provided
-      const signer = this.flags.execute
-        ? this.wallet.payer.publicKey
-        : new PublicKey(this.flags.signer || this.wallet.payer.publicKey)
+      // If execute flag is provided, a signer cannot, as the Keypair is required
+      const isInvalidSigner = !!this.flags.execute && !!this.flags.signer
+      this.require(!isInvalidSigner, 'To execute the transaction the signer must be loaded from a wallet')
+      const signer = new PublicKey(this.flags.signer || this.wallet.publicKey)
```

Comment:
Not executing but preparing a rawTx for a specific signer -  Do we even need this functionality anymore?

Reply:
I don't think we need it